library/kani/src/lib.rs

```diff
@@ -113,5 +113,13 @@ pub fn nondet<T: Arbitrary>() -> T {
 #[rustc_diagnostic_item = "KaniExpectFail"]
 pub fn expect_fail(_cond: bool, _message: &'static str) {}
 
+/// Function used to generate panic with a static message as this is the only one currently
+/// supported by Kani display.
+#[inline(never)]
+#[rustc_diagnostic_item = "KaniPanic"]
+pub fn panic(message: &'static str) -> ! {
+    panic!("{}", message)
```

Comment:
Is this line needed?

Reply:
We need something that never returns, so I just figured that it would be nice to have this statement for when we have the executable trace.

Reply:
I see. I got confused initially that there's mutual recursion (`panic` function calls macro and vice versa), but then realized that this is a diagnostic item, so the `panic` macro is not really getting called.

Reply:
`unreachable!` macro might make sense for this use case

Reply:
IMO `panic` makes sense, but maybe add a comment saying why it's not infinite recursion. (Obvious, I think, after pointing it out, but looks like an easy confusion to me.)